Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/App.tsx

```diff
@@ -14,12 +73,31 @@ const App = () => {
         <Header />
         <form name="simulator" className="flex w-full flex-col items-center">
           <div className="mb-4 flex w-full flex-col items-center gap-y-4">
-            <SpeedInput />
-            <BatteryInput />
+            <SpeedInput value={speed} onChange={handleSpeedChange}/>
+            {speedError && <div className="text-red-500">{speedError}</div>} {/* speed error message */}
+
+            <BatteryInput value={battery} onChange={handleBatteryChange}/>
+            {batteryError && <div className="text-red-500">{batteryError}</div>} {/* Battery error message */}
+          
           </div>
           <div className="flex w-full flex-row justify-center gap-4">
             <WeatherInput />
```

Comment:
your weather component needs to be a controlled input as well. Right now I can't change the weather input